test/api_server_unittest.cc

```diff
@@ -0,0 +1,78 @@
+#include "../src/api_server.h"
+#include "../src/configuration.h"
+#include "../src/store.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+class ApiServerTest : public ::testing::Test {
+ protected:
+  using Dispatcher = MetadataApiServer::Dispatcher;
+  std::unique_ptr<Dispatcher> CreateDispatcher(
+      const std::map<std::pair<std::string, std::string>,
+                     std::function<void()>>& handlers) {
+    Dispatcher::HandlerMap handler_map;
+    for (const auto& element : handlers) {
+      std::function<void()> handler = element.second;
+      handler_map.emplace(element.first, [handler](
+          const MetadataApiServer::HttpServer::request& request,
+          std::shared_ptr<MetadataApiServer::HttpServer::connection> conn) {
+        handler();
+      });
+    }
+    return std::unique_ptr<Dispatcher>(new Dispatcher(handler_map, false));
+  }
+
+  void InvokeDispatcher(
+      const std:: unique_ptr<Dispatcher>& dispatcher,
```

Comment:
Sigh, I thought I've caught them all... That's what I get for typing in code on my phone.
Let's remove the space after the `::`.